test/e2e/declare_consumer.test.ts

```diff
@@ -1,68 +1,82 @@
+    streamName = createStreamName()
+    nonExistingStreamName = createStreamName()
+    await rabbit.createStream(streamName)
+    publisher = await createPublisher(streamName, connection)
   })
 
   afterEach(async () => {
-    await connection.close()
-    await rabbit.deleteStream(testStreamName)
+    try {
+      await connection.close()
+      await rabbit.deleteStream(streamName)
+      await rabbit.closeAllConnections()
+      await rabbit.deleteAllQueues({ match: /my-stream-/ })
+    } catch (e) {}
   })
 
-  it("declaring a consumer on an existing stream - the consumer should handle the message", async () => {
+  it("should handle the message", async () => {
     const messages: Buffer[] = []
-    const publisher = await connection.declarePublisher({ stream: testStreamName })
-    await publisher.send(Buffer.from("hello"))
-
-    await connection.declareConsumer({ stream: testStreamName, offset: Offset.first() }, (message: Message) =>
+    await connection.declareConsumer({ stream: streamName, offset: Offset.next() }, (message: Message) =>
       messages.push(message.content)
     )
 
+    await publisher.send(Buffer.from("hello"))
+
     await eventually(() => expect(messages).eql([Buffer.from("hello")]))
   }).timeout(10000)
 
   it("declaring a consumer on an existing stream - the consumer should be handle more then one message", async () => {
     const messages: Buffer[] = []
-    const publisher = await connection.declarePublisher({ stream: testStreamName })
-    await publisher.send(Buffer.from("hello"))
-    await publisher.send(Buffer.from("world"))
-
-    await connection.declareConsumer({ stream: testStreamName, offset: Offset.first() }, (message: Message) =>
+    await connection.declareConsumer({ stream: streamName, offset: Offset.next() }, (message: Message) =>
       messages.push(message.content)
     )
 
-    await eventually(() => expect(messages).eql([Buffer.from("hello"), Buffer.from("world")]))
+    await publisher.send(Buffer.from("hello"))
+    await publisher.send(Buffer.from("world"))
+
     await eventually(() => expect(messages).eql([Buffer.from("hello"), Buffer.from("world")]))
   }).timeout(10000)
 
+  it(`consume a lot of messages`, async () => {
+    const receivedMessages: Buffer[] = []
+    //const receivedClassicMessages: Buffer[] = []
+
+    await connection.declareConsumer({ stream: streamName, offset: Offset.next() }, (message: Message) => {
+      receivedMessages.push(message.content)
+    })
+
+    const messages = range(1000).map((n) => Buffer.from(`hello${n}`))
+    for (const m of messages) {
+      await publisher.send(m)
+    }
+
+    // const { conn, ch } = await createClassicConsumer(streamName, (m) => receivedClassicMessages.push(m.content))
+    // await eventually(() => expect(receivedClassicMessages).eql(messages), 7000)
+    // await ch.close()
+    // await conn.close()
+    await eventually(() => expect(receivedMessages).eql(messages), 10000)
+  }).timeout(50000)
```

Comment:
Can you remove commented code?
Can I suggest to declare also the consumer after the publishing?